third_party/amd/lib/TritonAMDGPUTransforms/RemoveLayoutConversions.cpp

```diff
@@ -204,10 +204,12 @@ static bool hasConvertToAmdMmaTransisitiveUse(Operation *op,
     getForwardSlice(currentValue, &forwardSlice);
     for (Operation *op : forwardSlice) {
       if (auto convertOp = dyn_cast<triton::gpu::ConvertLayoutOp>(op)) {
-        if (convertOp.getResult()
-                .getType()
-                .cast<RankedTensorType>()
-                .getEncoding() == encoding)
+        Attribute dstEncoding = convertOp.getResult()
```

Comment:
@zhanglx13 this fixes IR issues (and accuracy issues with BLOCK_M = 128), which originated because mfma layout of the first dot was not propagated properly in remove layout conversion pass. I don't really understand the intent behind hasConvertToAmdMmaTransisitiveUse and it's usage in initAnchorLayout. Git blame show @micmelesse as the author. 
This is the comment I found:

          // Workaround to not propagate MFMA layout in case there are
          // no chained dots MFMA layout is expensive to convert, so we want
          // to convert it to something else as soon as possible.
          // It saves LDS space in some cases.
          //
          // TODO: rework this heuristic if we can store MFMA layout directly
          // into global memory.
          //
          // Also relevant for WMMA layout.

I don't understand "MFMA layout is expensive to convert" part and how does that change in chain dot case. How does it saves LDS?

Reply:
This part is our legacy from rocm fork... i have not changed the logic.
I think, it is not a rootcause.
I created a draft with a fix for wmma, as far as I can't test on mi arch: https://github.com/openai/triton/pull/3584

Reply:
Since you are here, can you remove the `USE_ROCM` at line 192?